extensions/src/AWSSDK.Extensions.NETCore.Setup/ClientFactory.cs

```diff
@@ -79,18 +79,18 @@ internal object CreateServiceClient(IServiceProvider provider)
         /// </summary>
         /// <param name="provider">The dependency injection provider.</param>
         /// <returns>The AWS service client</returns>
-        internal static IAmazonService CreateServiceClient(ILogger logger, Type serviceInterfaceType, AWSOptions options)
+        internal IAmazonService CreateServiceClient(ILogger logger, AWSOptions options)
         {
             PerformGlobalConfig(logger, options);
             var credentials = CreateCredentials(logger, options);
 
             if (!string.IsNullOrEmpty(options?.SessionRoleArn))
             {
-                credentials = new AssumeRoleAWSCredentials(credentials, options.SessionRoleArn, options.SessionName);
+                credentials = new AssumeRoleAWSCredentials(credentials, _awsOptions.SessionRoleArn, _awsOptions.SessionName);
             }
 
-            var config = CreateConfig(serviceInterfaceType, options);
-            var client = CreateClient(serviceInterfaceType, credentials, config);
+            var config = CreateConfig(options);
+            var client = CreateClient(credentials, config);
             return client as IAmazonService;
```

Comment:
Can we return `client as T` instead of  `client as IAmazonService;` and update this method return type?

Reply:
If I used generic `T` then AWSSDK.Extensions.NETCore.Setup would have to know the type which it can't because it doesn't have a dependency on all of our service packages.